compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -144,11 +145,31 @@ static FailureOr<Value> gpuAllocationFn(OpBuilder &builder, Location loc,
       .getResult();
 }
 
+// Checks that the 2 subviews are equivalent. Checks that all operands,
+// attributes, offsets, sizes, and strides are equal for both subviews.
+static bool areEquivalentSubviews(Value from, Value to) {
+  auto fromSubview = from.getDefiningOp<memref::SubViewOp>();
+  auto toSubview = to.getDefiningOp<memref::SubViewOp>();
+  if (!fromSubview || !toSubview) {
+    return false;
+  }
+  return llvm::equal(fromSubview->getOperands(), toSubview->getOperands()) &&
+         llvm::equal(fromSubview->getAttrs(), toSubview->getAttrs()) &&
+         llvm::equal(fromSubview.getMixedOffsets(),
+                     toSubview.getMixedOffsets()) &&
+         llvm::equal(fromSubview.getMixedSizes(), toSubview.getMixedSizes()) &&
+         llvm::equal(fromSubview.getMixedStrides(),
+                     toSubview.getMixedStrides());
```

Comment:
can you break this down into a couple boolean variables, so that this is a bit more debugable?

Reply:
Actually going a step further, I think just comparing `getOperands` and `getAttrs` is good enough, `getMixed*` is just an interleaving of dynamic and static things, which are both already checked with the attributes and operations. You might even be able to just compare `getAttributeDictionary()` instead of `getAttrs`, although I don't know if `DictionaryAttr` has special deduplication rules.